### DISCUSSION TITLE
You have already activated rake 13.0.3, but your Gemfile requires rake 13.0.6.

### DISCUSSION THREAD

**Discussion by dragon2611** [mastodon/mastodon#21313] [Q&A]
### Steps to reproduce the problem

Following https://github.com/McKael/mastodon-documentation/blob/master/Running-Mastodon/Docker-Guide.md with 3.53 (Also tried with cloning default branch)

get as far as 

`docker-compose run --rm web rake secret`

```
`Creating masterdon_web_run ... done
rake aborted!
Gem::LoadError: You have already activated rake 13.0.3, but your Gemfile requires rake 13.0.6. Prepending `bundle exec` to your command may solve this.
<internal:/opt/ruby/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
<internal:/opt/ruby/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
/opt/mastodon/config/boot.rb:8:in `<top (required)>'
/opt/mastodon/config/application.rb:1:in `require_relative'
/opt/mastodon/config/application.rb:1:in `<top (required)>'
<internal:/opt/ruby/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
<internal:/opt/ruby/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
/opt/mastodon/Rakefile:4:in `<top (required)>'
(See full trace by running task with --trace)
ERROR: 1`
```

Seems the bundle exec suggestion works in 3.53, Didn't work when I'd originally only cloned the repo without specifying a version

### Expected behaviour

secret generated

### Actual behaviour

error as above

### Specifications

masterdon 3.53
Docker.

**Comment by sean666888** [comment]
Same

**Comment by ghost** [comment]
Works for me when I run this way:

`docker-compose run --rm web bundle exec rake secret `

**Comment by decisoft** [comment]
> Works for me when I run this way:
> 
> `docker-compose run --rm web bundle exec rake secret `

Works for me too! Thanks

**Comment by rb-26** [comment]
Correct, this does give you the secret but it does not give you the rest of the config for the .env file

**Comment by ghost** [comment]
> Correct, this does give you the secret but it does not give you the rest of the config for the .env file

@rb-26 Is that not obvious though?

For those who want the whole .env configuration in one go:

`docker-compose run --rm web bundle exec rake mastodon:setup`

**Reply by ghost** [reply]
@nightpool Thanks for moving here!
